### PR TITLE
dateParser needs to handle invalid base dates

### PR DESCRIPTION
**Pull request by JohannesRudolph**
I noticed that datePicker would fail in the following scenario (reproducible with the official sample): 
 - clear input text
 - enter a valid date according to format
 - observe that the date is parsed as invalid date

The underlying issue seems to be that the dateParser is handled an invalid base date (as in: a Javascript InvalidDate object). This pull request fixes this scenario and should be trivial to review as it contains a failing test which passes with the suggested fix.

### CODE REVIEW

**Comment by bitstorm** [comment]
+1 I need it badly :)

File: src/dateparser/dateparser.js [hunk 1]
```diff
@@ -124,8 +124,10 @@ angular.module('ui.bootstrap.dateparser', [])
         results = input.match(regex);
 
     if ( results && results.length ) {
-      var fields, dt;
-      if (baseDate) {
+      var fields, dt,
+          isValidBaseDate = baseDate && !isNaN(baseDate.valueOf());
```
Comment (wesleycho) [inline src/dateparser/dateparser.js]: This doesn't look quite right - this should probably be `angular.isDate(baseDate) && !isNaN(baseDate.getTime())`.

**Comment by wesleycho** [comment]
I believe this PR is superceded by #3759 . Thoughts?

**Comment by rvanbaalen** [comment]
It is indeed @wesleycho

**Comment by chrisirhc** [comment]
Closing in favor of #3759

**Pull request closed without merging**